lxd/images.go

```diff
@@ -916,12 +916,99 @@ func getImageMetadata(fname string) (*api.ImageMetadata, string, error) {
 	return &result, imageType, nil
 }
 
-func doImagesGet(d *Daemon, recursion bool, project string, public bool) (interface{}, error) {
+func evaluateFieldImage(entry FilterEntry, image *api.Image) bool {
+	result := false
+	value := entry.Value
+	field := entry.Field
+
+	switch {
+		case strings.EqualFold(field, "architecture"):
+			result = value == image.Architecture
+			break
+
+		case strings.EqualFold(field, "filename"):
+			result = value == image.Filename
+			break
+
+		case strings.EqualFold(field,"type"):
+			result = image.Type == value
+			break
+
+		case strings.EqualFold(field,"fingerprint"):
+			result = image.Fingerprint == value
+			break
+
+		case strings.EqualFold(field,"public"):
+			imagePut := image.ImagePut
+			if strings.EqualFold(value,"true"){
+				result = imagePut.Public
+			} else {
+				result = !imagePut.Public
+			}
+			break
+
+		case strings.EqualFold(field,"autoupdate"):
+			imagePut := image.ImagePut
+			if strings.EqualFold(value,"true"){
+				result = imagePut.AutoUpdate
+			} else {
+				result = !imagePut.AutoUpdate
+			}
+			break
+
+		case strings.HasPrefix(field, "UpdateSource"):
+			fieldCut := field[13:len(field)]
+			source := image.UpdateSource
+			switch (fieldCut) {
+				case "Alias":
+					result = source.Alias == value
+					break
+				case "Certificate":
+					result = source.Certificate == value
+					break
+				case "Protocol":
+					result = source.Protocol == value
+					break
+				case "Server":
+					result = source.Server == value
+					break
+				case "ImageType":
+					result = source.ImageType == value
+					break
+				default:
+					result = false
+					break
+			}
+			
+			break
+
+		case strings.HasPrefix(field, "Properties"):
+			fieldCut := field[11:len(field)]
+			imagePut := image.ImagePut
+			result = value == imagePut.Properties[fieldCut]
+			break
```

Comment:
Same as for instances, it would be good to not have to hardcode the entire structure of the struct in there but instead use reflecting to match the filter keys with the struct members.